### ISSUE TITLE
edge case to the rule `unicorn/explicit-length-check`

### ISSUE THREAD

**Issue by shaffy9633** [sindresorhus/eslint-plugin-unicorn#2595]
So for the rule `unicorn/explicit-length-check` it does not account for the case of if an optional chaining operator is present. 

`unicorn/explicit-length-check`

So for example if I write 
```js
if (!x.length) {
some logic
}
```
it flags and auto changes to 
```js
if (x.length === 0) {
some logic
}
```
but if it is 
```js
if (!x?.length) {
some logic
}
```
it should really flag and change to 

```js
if (!x || x.length === 0) {
some logic
}
```

the same goes for the positive case, in which case the fix would be `x && x.length > 0`
